Controllers/ProductsController.cs

```diff
@@ -21,57 +22,56 @@ public ProductsController(IProductService productService)
         public async Task<ActionResult<List<ProductViewModel>>> GetAllProducts() 
         {
             List<ProductViewModel> listOfProductViewModels = new List<ProductViewModel>();
-            
-            List<Product> listOfProducts = await _productService.GetAll();
+            ProductViewModel viewModel;
+
+            List<Product> listOfProducts = await _productService.GetAllAsync();            
 
             foreach(Product p in listOfProducts) 
             {
-                listOfProductViewModels.Add(new ProductViewModel()
-                {
-                    Id = p.Id,
-                    Name = p.Name,
-                    Price = p.Price
-                });
+                viewModel = ProductViewModel.MapFromProduct(p);
+
+                listOfProductViewModels.Add(viewModel);
```

Comment:
I would suggest to rewrite it using LINQ instead of creating new list and iterating with foreach, it can be just one line